src/pages/instances/actions/RestartInstanceBtn.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import ConfirmationButton from "components/ConfirmationButton";
+import { restartInstance } from "api/instances";
+import { useNotify } from "context/notify";
+import { CheckboxInput } from "@canonical/react-components";
+import { useInstanceLoading } from "context/instanceLoading";
+import InstanceLink from "pages/instances/InstanceLink";
+
+interface Props {
+  instance: LxdInstance;
+}
+
+const RestartInstanceBtn: FC<Props> = ({ instance }) => {
+  const instanceLoading = useInstanceLoading();
+  const notify = useNotify();
+  const [isForce, setForce] = useState(false);
+  const queryClient = useQueryClient();
+  const isLoading =
+    instanceLoading.getType(instance) === "Restarting" ||
+    instance.status === "Restarting";
+
+  const handleRestart = () => {
+    instanceLoading.setLoading(instance, "Restarting");
+    restartInstance(instance, isForce)
+      .then(() => {
+        notify.success(
+          <>
+            Instance <InstanceLink instance={instance} /> restarted.
+          </>
+        );
+      })
+      .catch((e) => {
+        notify.failure(
+          <>
+            Error instance restart for <InstanceLink instance={instance} />.
+          </>,
+          e
+        );
+      })
+      .finally(() => {
+        instanceLoading.setFinish(instance);
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances],
+        });
+      });
+  };
+
+  const disabledStatuses = ["Stopped", "Frozen"];
+  const isDisabled = isLoading || disabledStatuses.includes(instance.status);
```

Comment:
This would allow the user to restart an instance that has the "error" status. Is this actually admissible?

Reply:
The reason is that sometimes a normal stop gets stuck. And I wanted to keep a way to then trigger a force stop.

Reply:
I see, should we then add the "Error" status to the `disabledStatuses` maybe? Just to be sure that we don't allow something that should not be allowed...

Reply:
I misunderstood your first comment. Yes - that makes sense.